framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -175,6 +187,115 @@ private void updateInferredExecutableParameterTypes(
         }
     }
 
+    @Override
+    public void updateContracts(
+            Analysis.BeforeOrAfter preOrPost,
+            ExecutableElement methodElt,
+            AnnotatedTypeFactory atf,
+            CFAbstractStore<?, ?> store) {
+
+        if (store == null) {
+            throw new BugInCF(
+                    "updateContracts(%s, %s, %s, null)",
+                    preOrPost, methodElt, atf.getClass().getSimpleName());
+        }
+
+        String className = getEnclosingClassName(methodElt);
+        String jaifPath = storage.getJaifPath(className);
+        AClass clazz =
+                storage.getAClass(className, jaifPath, ((MethodSymbol) methodElt).enclClass());
+        AMethod amethod = clazz.methods.getVivify(JVMNames.getJVMMethodSignature(methodElt));
+        amethod.setFieldsFromMethodElement(methodElt);
+
+        // TODO: Probably move some part of this into the AnnotatedTypeFactory.
+
+        // TODO: Handle more than just fields of "this".
+
+        // Process the store's fields.
+        for (Map.Entry<FieldAccess, ? extends CFAbstractValue<?>> entry :
+                store.getFieldValues().entrySet()) {
+            FieldAccess fa = entry.getKey();
+            CFAbstractValue<?> v = entry.getValue();
+            VariableElement fieldElement = fa.getField();
+            AnnotatedTypeMirror fieldType = atf.getAnnotatedType(fieldElement);
+
+            AnnotatedTypeMirror atm = convertCFAbstractValueToAnnotatedTypeMirror(v, atf);
```

Comment:
`convertCFAbstractValueToAnnotatedTypeMirror` creates a partially-annotation AnnotatedTypeMirror which can cause crashes if passed to isSubtype or leastUpperBound or anywhere the framework expects fully annotated types.  Instead,  `convertCFAbstractValueToAnnotatedTypeMirror` should make a copy of `fieldType` and change its primary annotations to those in `v`.